connman/plugins/sailfish_ofono.c

```diff
@@ -582,13 +599,102 @@ static int modem_configure(struct modem_data *md)
 	return index;
 }
 
+static gboolean is_modem_network_configured(struct modem_data *md)
+{
+	DBG("%p", md);
+
+	switch (md->connctx->protocol) {
+	case OFONO_CONNCTX_PROTOCOL_UNKNOWN:
+	case OFONO_CONNCTX_PROTOCOL_NONE:
+		return FALSE;
+	case OFONO_CONNCTX_PROTOCOL_IP:
+		if (!connman_network_is_configured(md->network,
+						CONNMAN_IPCONFIG_TYPE_IPV4)) {
+			connman_warn("ofono: %p IPv4 no address set", md);
+			return FALSE;
+		}
+
+		break;
+	case OFONO_CONNCTX_PROTOCOL_IPV6:
+		if (!connman_network_is_configured(md->network,
+						CONNMAN_IPCONFIG_TYPE_IPV6)) {
+			connman_warn("ofono: %p IPv6 no address set", md);
+			return FALSE;
+		}
+
+		break;
+	case OFONO_CONNCTX_PROTOCOL_DUAL:
+		if (!connman_network_is_configured(md->network,
+						CONNMAN_IPCONFIG_TYPE_IPV4)) {
+			connman_warn("ofono: %p DUAL no IPv4 address set", md);
+			return FALSE;
+		}
+
+		if (!connman_network_is_configured(md->network,
+						CONNMAN_IPCONFIG_TYPE_IPV6)) {
+			connman_warn("ofono: %p DUAL no IPv6 address set", md);
+			return FALSE;
+		}
+
+		break;
+	}
+
+	DBG("%p is configured", md);
+
+	return TRUE;
+}
+
+static gboolean delayed_connect(gpointer data) {
```

Comment:
How about naming it `modem_delayed_connect()` or something, to reduce the number of prefixes used in this file. And the opening bracket should be on the next line.

Reply:
Yeap, makes it more readable. Changed.